Jdp190401 40 naprawa problemu z bazami danych

pszczepcio opened the pull request:
- usuniecie @Transactional z Dao
- zamiana Eager na Lazy w relacji 1:N
- dodanie @Transactional do ProductTestSuite

pszczepcio committed:
- usunienie adnotacji @Transactional z Dao
- zamiana Eager na Lazy w relacji 1:N
- dodanie @Transactional do ProductTestSuite

KucharM reviewed (commented):
mam pytanie co powoduje że usuwamy @Transactional z repozytorium a dodajemy nad testy?

KucharM commented:
czy w przypadku późniejszych implementacji metod w repozytorium @Transactional nie trzeba będzie przywrócić?

Makedo81 reviewed (commented):
Rozumiem ,ze usuwamy @Transactional w Dao a adnotacja ta bedzie  umieszczona w Service.

Makedo81 commented:
> 
> 
> mam pytanie co powoduje że usuwamy @transactional z repozytorium a dodajemy nad testy?

Chodzi o to, zeby wycofac albo wprowadzic dane do bazy danych w trakcie wykonywania testow? Jezeli tak, to dlaczego koliduje to z @Transactional umieszczonym w repozytorium?

KucharM commented:
> Rozumiem ,ze usuwamy @transactional w Dao a adnotacja ta bedzie umieszczona w Service.

wydaje mi się że @Transactional w ogóle nie powinno być w repozytorium

pszczepcio commented:
Nie potrafię odpowiedźieć na Wasze pytanie gdzie dokładnie ma być @Transactional. Najważniejsze, że teraz działa. Może Karol coś nam więcej powie na ten temat.

klikuskodilla committed:
Merge branch 'master' into JDP190401-40

klikuskodilla commented:
Ta anotacja jest opcjonalna, dodaje się ją np gdy wywołujemy kod wieloetapowo, tutaj nie ma takiej potrzeby. W testach prawdopodobnie też jest opcjonalna

pszczepcio committed:
Merge branch 'master' into JDP190401-40

pszczepcio committed:
- aktualizacja kodu o userTesTSuite
- musiałem dołożyć adnotację @Transactional w UsertTestSuite
- w userTestSuite musiałem usunąć cartDao. Jest automatycznie usuwane przez orderDao.deletebyId()

pszczepcio committed:
Merge remote-tracking branch 'origin/JDP190401-40' into JDP190401-40

Makedo81 reviewed (commented):
A co dokladnie ci nie dziala, ze dodajesz Transactional nad testy. Ja nie dodawalam w moich.

pszczepcio commented:
Szczerze mówiąc nie sprawdzałem czy program działa jak nie mam w testach @Transactional. Za chwilę sprawdzę i dam odpwiedź na grupowym kanale.

Makedo81 reviewed (dismissed):
Faktycznie aplikacja sie uruchamia jezeli usuniemy @Transactional i zamiana Eager na Lazy.W przeciwnym wypadku dzialaja tylko testy

klikuskodilla committed:
Merge branch 'master' into JDP190401-40

pszczepcio committed:
Merge branch 'master' into JDP190401-40

pszczepcio committed:
Merge remote-tracking branch 'origin/JDP190401-40' into JDP190401-40

klikuskodilla committed:
Merge branch 'master' into JDP190401-40

pszczepcio committed:
Merge branch 'master' into JDP190401-40

pszczepcio committed:
- musiałem zakomentować productDao.deleteById encja jest usuwana razem z encja cart
- zakomentowałem dwie linijki kodu przez ktory nie wychodziły testy. Do listy produktów był dodawany produkt a nastepnie lista była zerowana przez pusta arrayliste

pszczepcio committed:
- musiałem zakomentować productDao.deleteById encja jest usuwana razem z encja cart
- zakomentowałem dwie linijki kodu przez ktory nie wychodziły testy. Do listy produktów był dodawany produkt a nastepnie lista była zerowana przez pusta arrayliste

pszczepcio committed:
Merge remote-tracking branch 'origin/JDP190401-40' into JDP190401-40

klikuskodilla reviewed (changes requested):
Dlaczego komentujesz kod zamiast usunąć źródło problemu?

pszczepcio commented:
Nie wiedziałem czy mogę od razu usuwać czy nie. To zaraz poprawię to jeszcze raz.

klikuskodilla commented:
Jeśli to jest poprawka nie związana z tym taskiem, możesz stworzyć nowy PR który to naprawia, i zaznaczyć tutaj że aktualny czeka na zmergowanie tego drugiego, dzięki czemu tutaj nie trzeba dodawać żadnego kodu.

pszczepcio committed:
- poprawa testów OrderDao w metodzie findAll
- usunięcie zakomentowanych częsci kodu będących błędami

klikuskodilla commented:
problem dla innych deweloperów jest taki że niby coś przestało działać, nie do końca wiadomo co wiec zakomentowałeś, teraz odkomentowałeś i tak naprawdę nie wiadomo co nie działało i czy przyczyna została usunięta

pszczepcio commented:
To spróbuję rozdzielić to na dwa taski. a może ten task obecny zamknąc i zrobić nowy żeby to było czytelniejsze?
